File: domain/src/main/java/com/arturogutierrez/openticator/domain/password/interactor/CreateMasterPasswordInteractor.java
package com.arturogutierrez.openticator.domain.password.interactor;

import com.arturogutierrez.openticator.domain.Preferences;
import com.arturogutierrez.openticator.domain.password.PasswordSerializer;

public class CreateMasterPasswordInteractor {

  private final Preferences preferences;
  private final PasswordSerializer passwordSerializer;

  public CreateMasterPasswordInteractor(Preferences preferences,
      PasswordSerializer passwordSerializer) {
    this.preferences = preferences;
    this.passwordSerializer = passwordSerializer;
  }

  public void createMasterPassword(String plainPassword) {
    String encodedPassword = passwordSerializer.encodePassword(plainPassword);

    if (encodedPassword == null) {
      // it's a bad practice store the password in plain
      // but this should not happen (it's just in case)
      encodedPassword = plainPassword;
    }

    preferences.setMasterPassword(encodedPassword);
  }
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/password/interactor/CreateMasterPasswordInteractor.kt
package com.arturogutierrez.openticator.domain.password.interactor

import com.arturogutierrez.openticator.domain.password.PasswordSerializer
import com.arturogutierrez.openticator.storage.preferences.Preferences

class CreateMasterPasswordInteractor(
    private val preferences: Preferences,
    private val passwordSerializer: PasswordSerializer) {

  fun createMasterPassword(plainPassword: String) {
    var encodedPassword = passwordSerializer.encodePassword(plainPassword)

    if (encodedPassword == null) {
      // it's a bad practice store the password in plain
      // but this should not happen (it's just in case)
      encodedPassword = plainPassword
    }

    preferences.masterPassword = encodedPassword
  }
}
